### ISSUE TITLE
add custom transition support

### ISSUE THREAD

**Issue by csells** [csells/go_router#5]
to enable using go_router as a complete replacement for Nav2

**Comment by csells** [comment]
leverage the builders?

```dart
MaterialApp(
  theme: ThemeData(
      pageTransitionsTheme: const PageTransitionsTheme(
        builders: <TargetPlatform, PageTransitionsBuilder>{
          TargetPlatform.android: FadeUpwardsPageTransitionsBuilder(),
          TargetPlatform.iOS: CupertinoPageTransitionsBuilder(),
          TargetPlatform.linux: FadeUpwardsPageTransitionsBuilder(),
          TargetPlatform.macOS: CupertinoPageTransitionsBuilder(),
          TargetPlatform.windows: FadeUpwardsPageTransitionsBuilder(),
        },
      )
  )
)
```

**Comment by csells** [comment]
and a potential helper page: https://github.com/flutter/flutter/issues/88221#issuecomment-918576637

**Comment by csells** [comment]
fixed w/ v0.9.5
